### PR TITLE
ci: fix Docker container build

### PR DESCRIPTION
**Pull request by fhunleth**

### CODE REVIEW

**Review by jjcarstens — COMMENTED** [review]
Seems like this isn't quite fixed?

**Comment by fhunleth** [comment]
No, I pinged Digit for help.

File: .circleci/config.yml [hunk 1]
```diff
@@ -33,8 +33,10 @@ docker_build: &docker_build
   run:
     name: Build docker images
     command: |
-      docker buildx create --use
-      docker buildx build --platform linux/arm64,linux/amd64 $DOCKER_REPO:$DOCKER_TAG $DOCKER_PATH
+      apk add --no-cache docker-cli-buildx
+      docker context create my_context
+      docker buildx create --use my_context
+      docker buildx build --platform linux/arm64,linux/amd64 $DOCKER_PATH
```
Comment (doawoo) [inline .circleci/config.yml:39]:
```suggestion
       apk add --no-cache docker-cli-buildx
       docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
       docker buildx create --name multiarch --driver docker-container --use
       docker buildx build --platform linux/arm64,linux/amd64 $DOCKER_PATH
```

There's apparently a bug in Docker... https://github.com/docker/buildx/issues/495

File: .circleci/config.yml [hunk 2]
```diff
@@ -33,10 +33,10 @@ docker_build: &docker_build
...
     name: Build docker images
     command: |
-      apk add --no-cache docker-cli-buildx
-      docker context create my_context
-      docker buildx create --use my_context
-      docker buildx build --platform linux/arm64,linux/amd64 $DOCKER_PATH
+       apk add --no-cache docker-cli-buildx
+       docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+       docker buildx create --name multiarch --driver docker-container --use
+       docker buildx build --platform linux/arm64,linux/amd64 $DOCKER_PATH
```
Comment (doawoo) [inline .circleci/config.yml:39]:
```suggestion
       apk add --no-cache docker-cli-buildx
       docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
       docker context create multiarch
       docker buildx create multiarch --driver docker-container --use
       docker buildx build --platform linux/arm64,linux/amd64 $DOCKER_PATH
```

MAYBE this one?? CI debugging is painful :(

**Review by doawoo — CHANGES_REQUESTED** [review]

**Comment by doawoo** [comment]
OK! I think I have finally figured out the dark arts of creating a multi-arch docker image. Turns out you cannot just push one easily outside of using `--push` in the `buildx` sub-command. So I build them individually, push them as dev tags, then combine them into a single `manifest` which will act like a "unified" AMD64/ARM64 image when pulled.

NOTE: we cannot test the ARM image easily in CircleCI, they do not support Docker ARM execution very well, and, it would simply take forever...

**Comment by fhunleth** [comment]
@doawoo You're a hero! Thank you for figuring this out!!!!!

**Pull request merged**